### ISSUE TITLE
PeriodGranularity doesn't work right in Kathmandu

### ISSUE THREAD

**Issue by gianm** [apache/incubator-druid#4073]
Test case below. PT1H works fine but PT2H doesn't.

```java
@Test
public void testTruncateKathmandu() throws Exception
{
  final DateTimeZone tz = DateTimeZone.forID("Asia/Kathmandu");
  final DateTime date = new DateTime("2011-03-15T21:42:23.898+05:45").withZone(DateTimeZone.UTC);
  final PeriodGranularity year = new PeriodGranularity(new Period("P1Y"), null, tz);
  final PeriodGranularity hour = new PeriodGranularity(new Period("PT1H"), null, tz);
  final PeriodGranularity twoHour = new PeriodGranularity(new Period("PT2H"), null, tz);

  // These two pass
  Assert.assertEquals(
      new DateTime("2011-01-01T00:00:00.000+05:45", tz),
      year.toDateTime(year.bucketStart(date).getMillis())
  );
  Assert.assertEquals(
      new DateTime("2011-03-15T21:00:00.000+05:45", tz),
      hour.toDateTime(hour.bucketStart(date).getMillis())
  );

  // This one fails
  Assert.assertEquals(
      new DateTime("2011-03-15T20:00:00.000+05:45", tz),
      twoHour.toDateTime(twoHour.bucketStart(date).getMillis())
  );
}
```

**Comment by gianm** [comment]
I hope I'm not being too optimistic with the easy tag :)

**Comment by jihoonson** [comment]
Fixed in #5410.
